### ISSUE TITLE
Anki freezes when trying to copy one card

### ISSUE THREAD

**Issue by galantra** [Arthur-Milchior/anki-copy-note#5]
Since the last update, Anki freezes when I try to copy any card.

Please tell me what further information could be useful, I'll supply it.

**Comment by Arthur-Milchior** [comment]
I can't reproduce this. Can you please let me know what is your current configuration. (From add-on manager, select «copy note», click on «config»). 

Can you also please let me know what other add-on you have. A screen shot would be sufficient. It may helps figure out whether another add-on is interacting.

Better, can you share with me the notes you want to copy. You can use add-on https://ankiweb.net/shared/info/1983204951 to select them and export those notes. This way, I'll see whether there is something particular with your notes which cause the problem.

**Comment by galantra** [comment]
[Some notes](https://send.firefox.com/download/9e13bf3651eed634/#I2ipqfBeKfJ-KC-RKDI7GA
)
{
    "Copy log": true,
    "Preserve creation time": true,
    "Preserve ease, due, interval...": true,
    "Shortcut: copy": "Ctrl+C"
}

Anki 2.1.15 (6661ba8a) Python 3.6.7 Qt 5.12.1 PyQt 5.11.3
Platform: Windows 10
Flags: frz=True ao=True sv=1

Add-ons:

Extended Card Stats During Review
TagSelectorV2 (disabled)
Pennywise Add Only Once Option (disabled)
extended html editor
Speed Focus Mode auto-alert auto-reveal auto-fail (disabled)
ReMemorize Ex Rated0Search
Color Confirmation (disabled)
put ALL due learning cards first fork for 21 (disabled)
Newline in strings in add-ons configurations
Search in formatted text in any fields (disabled)
Straight Reward Ease Factor grows at 5 straight Good rating in review
Improve speed of change of note type
Pretzel Logic Retention Benchmarking (disabled)
Convert Subdecks to Tag Hierarchy (disabled)
HoochiePapa Randomize New Queue
Extended Schema Hash
ImageResizer
Add Table
Field History (disabled)
Change cards decks prefix (disabled)
Correcting a bug in anki which makes new card appearing in wrong order
Insert Randomized Lists multiple choice
Close and lose current input  for sticky fields (disabled)
Cacher In The Rye Windows
Quick note and deck buttons Fork for 21 (disabled)
CTRLF5 to Refresh the Browser
Tag Entry Enhancements
Image Occlusion Enhanced for Anki 21 alpha
Search and Replace Tags
Delete empty NEW cards
A half-automatic Forvo Downloader (disabled)
Fluent Forever - Open Websites (disabled)
HoochieMama Randomize Rev Queue
Symbols (disabled)
Another Retreat go upstream the history for the next interval at lapse (disabled)
Night Mode
Cookie monster (disabled)
GODMODE faster shortcuts and Cloze switching (disabled)
Cardistry Dynamically Adjust New Cards
Pop-up Dictionary beta (disabled)
Reviewer to Browser choosing what to display
OpenInExternalEditorRenameDuplicate for ImageAudioVideo
Copy notes
Full Screen Toggle (disabled)
Advanced Card Styles (disabled)
Touch Screen - draw and write on your cards (disabled)
Refocus Card when Reviewing 21
MIA Retirement Addon (disabled)
More consistent cards generation (disabled)
Agent Orange Pseudo-Fuzz Defoliator
Hoarders Delight RecycleTrash Bin
Always On Top (disabled)
Search in Add Card Dialog
CrowdAnki JSON exportimport Edit history Collaborate on deck creation
Push Paradox Sibling Cardistry
Fast Word Query Multi-threaded queries for words from local or web dictionaries (disabled)
Interactive Cards for language learning
Browser MaximizeHide TableEditor (disabled)
Stats Overview Pie Graph with Distinct Learning and-or Relearning Sections (disabled)
Large and Colorful Buttons (disabled)
Anki Excel Sync (disabled)
Quick  Easy (disabled)
auto rate typed answer (disabled)
movie2anki (disabled)
Zoom 21
HoochieBaby Queue Controller  Day-Lrn Q
Update add-ons when collection is synchronized (disabled)
Add Audio and Pronunciation from Merriam-Webster (disabled)
Feed to Anki for Word of the Day RSS (disabled)
Advanced Copy Fields
Ignore accents in browser search (disabled)
Fill the blanks - Multiple typecloze support (disabled)
Trigger and actions change one card depending on what occurs on another card (disabled)
Export cards selected in the Browser
Customize Sidebar (disabled)
Open linked pdf docx epub audiovideo etc in external Program (disabled)
If a note has no more card warns instead of deleting it
Add a tag to notes with missing media (disabled)
remove linebreaks  do word wrap fork for 21 (disabled)
Five Finger Silver New Answer Button Graph ReLrn Stats ReMemorize Plan9-0 (disabled)
AnkiConnect
Enhanced cloze 21 (disabled)
Ensure each note is unique correct a bug introduced by add-on copy note
Load Balanced Scheduler
Le Petit Debugger (disabled)
Automatic Basic to Cloze
Alternative Dialog for ReMemorize Touch-Friendly (disabled)
AutoDefine - Automatically define vocabulary words with pronunciations and image (disabled)
Card Info Bar for Browser (disabled)
Card Info During Review (disabled)
Highlight Search Results in the Browser
Maximum image height in card editor (disabled)
Customize Keyboard Shortcuts
Empty cards returns more usable informations
Progress Graphs and Stats for Learned and Matured Cards (disabled)
Do not underline current queue in reviewer
Batch Editing
Mini Format Pack
Quick tagging 21
Cross Profile Search and Import
Add Hyperlink
3ft Under Bury Newly Added Cards
ReMemorize Rescheduler with sibling and logging v114
Opening the same window multiple time
Reviewer Context Menu Search (disabled)
Holdem Cardfield Preserves add card history
Clickable Tags v10 (disabled)
HTML Editor Tweaks v01 (disabled)
Bury related notes
Keep model of add cards (disabled)
Allows empty first field during adding and import (disabled)
Throughput Historical (disabled)
Ease Factor Histogram (disabled)
Replay buttons on card
Improving change note type
Frozen Fields
Long term backups
Removes Empty Note Types for 21
Improved Quizlet to Anki 21 Importer (disabled)
Batch Download Pictures From Google Images (disabled)
add-on window search bar
Blitzkrieg Advanced Browser Sidebar BETA4
Web Query  (disabled)
avgEase
Numeric Keypad Remapping  ease bugs fixed
Syllabus - Anki Statistics and Export by Tag and Deck (disabled)
deck and card info sidebar during review (disabled)
Clean the default deck (disabled)
Deck name in title 21
Life Drain (disabled)
Import Video Add new cards from a video with subtitles (disabled)
Polar Connect - PDF and Incremental Reading with Anki Sync (disabled)
More Overview Stats 21 (disabled)
Additional Card Fields Fork for 21 (disabled)
Small Potatoes Show learning step count in days (disabled)
copy note contents to new Add window from Browser Reviewer Add EditCurrent
3 add-ons merged quicker anki explain deletion explain database check
Increase autosave frequency (disabled)
Anki IPA (disabled)
Adding note and changing note type become quicker
extended table editor
Pomodore Tomato Clock Port (disabled)
AwesomeTTS Google Cloud Text-to-Speech unofficial
Throughput Monitor (disabled)
Echelon - Improved Hierarchical Tags for Anki 21 (disabled)
Minimize to tray (disabled)
Create Filtered Deck from the Browser (disabled)
Reflect On Mee - Helps Alleviate RGS (disabled)
Open Added Today from Reviewer (disabled)
Web Browser - Search selection Import texts and images automatically (disabled)
Advanced Browser
Enhance main window
Slackers Delight PostponeDefer Button
Knockout Deck Subset Learning
MorphMan for Anki 21
Undelete deleted note (disabled)
Preview any cloze number
True Retention by Card Maturity
Editor Autocomplete for Anki 21 (disabled)
Incremental Reading v4103 (disabled)
Text Rendering Fix (disabled)
Correct Answers Graph (disabled)
No Penalties or Boosting (disabled)
PassFail (EN) (disabled)
ResetEZ (disabled)
autoLearningSteps (disabled)
bettertags (disabled)
Cloze Overlapper
more_answer_buttons
pez_dispenser (disabled)
rememorize_btn (disabled)
Review Heatmap (disabled)
roommate_switch (disabled)
show_warning_if_no_tags_when_adding_cards
Spell Checker (disabled)
tagedit_subtag_completer
touch_wood
visual_feedback
wordsworth (disabled)

**Comment by Arthur-Milchior** [comment]
I did import the cards you shared; I can copy them without any trouble.

I have never seen such a big list of add-ons. I have no idea how I could ever try to figure out with which other add-on(s) my add-on is interacting. Because I easily believe that the problem comes from the fact that two add-ons interacts badly. I don't want to take time to figure uot all of their add-on numbers and install them all. 

Do you know the principle of dichotomic test ? I guess that it would be the easiest way to figure out which add-on cause the trouble.
You can do the test by moving add-ons out of the add-on folder, and then back in it. I believe it's easier than using «enable»/«disabel»; since it does not force you to pay attention to which add-ons are enabled/disabled.

**Comment by galantra** [comment]
Yes sure, I completely understand that you don't take that time.
I am not an expert in programming and therefore I am very thankful for your guidance.

I will move half of the add-ons out of the folder and see which half causes interence. Then I repeat the process with one half of that half, etc.

**Comment by TheVinhLuong** [comment]
I am facing the same problem

**Comment by Arthur-Milchior** [comment]
I'm really trying to understand what is going on here. It makes absolutly no sens that I've got no problem, and that three people have it; even if it's the only add-on.

@TheVinhLuong Can you please give me the version of anki that you use, and your operating system 
Better, if you can, could you send me your whole collection. May be if I have the exact collection, I can reproduce the bug.
The file I want is called collection.anki2 if you don't know where to find it, I can help you with that when I know which OS you have. Normally, I would not ask this, but I honestly can't think of anything else right now to try to reproduce your bug

**Comment by Arthur-Milchior** [comment]
Can you both give me the configuration of the add-on, please. (From add-on manager, select «copy note», click on «config»).

**Comment by galantra** [comment]
Here you go
```
{
    "Copy log": true,
    "Preserve creation time": true,
    "Preserve ease, due, interval...": true,
    "Shortcut: copy": "Ctrl+C"
}
```

**Comment by TheVinhLuong** [comment]
@Arthur-Milchior I set `Copy log`'s value to `false` in configuration and it is working fine now. So I think the issue lies there. But if you still need my colllection.anki2 then you can get it [here](https://drive.google.com/file/d/1RdChZhBWIBAIUGxf8c67CFoJJHDwI8XJ/view?usp=sharing)
My OS: Window 10 Pro, version 1903, OS build 18362.356
Anki: Version 2.1.15 (442df9d6)
@galantra Please confirm if this workaround worked.

**Comment by galantra** [comment]
I confirm the workaround worked for me.

**Comment by TheVinhLuong** [comment]
@galantra Yeah, I just want to know (maybe Arthur too) if your issue and mine are of the same cause.

**Comment by Arthur-Milchior** [comment]
Would you be nice enough to update and set «copy log» to True; then tell me whether the problem is corrected or not. I hope it is.

**Comment by TheVinhLuong** [comment]
@Arthur-Milchior I need to restart Anki after update and set `Copy log` to `true`. It is working now, thank you.

**Comment by galantra** [comment]
The update has resolved my issue, too. I can copy notes again. Superb!
